Src/DependencyCollector/Microsoft.AI.DependencyCollector/DependencyCollectorDiagnosticListener.cs

```diff
@@ -0,0 +1,235 @@
+            {
+                string httpMethod = request.Method.Method;
+                Uri requestUri = request.RequestUri;
+                string resourceName = requestUri.AbsolutePath;
+                if (!string.IsNullOrEmpty(httpMethod))
+                {
+                    resourceName = httpMethod + " " + resourceName;
+                }
+
+                DependencyTelemetry telemetry = new DependencyTelemetry();
+                this.client.Initialize(telemetry);
+                telemetry.Start();
+                telemetry.Name = resourceName;
+                telemetry.Target = requestUri.Host;
+                telemetry.Type = RemoteDependencyConstants.HTTP;
+                telemetry.Data = requestUri.OriginalString;
+                this.pendingTelemetry.TryAdd(loggingRequestId, telemetry);
+
+                HttpRequestHeaders requestHeaders = request.Headers;
+                if (requestHeaders != null)
+                {
+                    if (!string.IsNullOrEmpty(telemetry.Context.InstrumentationKey) && !ContainsRequestContextKeyValue(requestHeaders, RequestResponseHeaders.RequestContextSourceKey))
+                    {
+                        string sourceApplicationId;
+                        if (this.correlationIdLookupHelper.TryGetXComponentCorrelationId(telemetry.Context.InstrumentationKey, out sourceApplicationId))
+                        {
+                            SetRequestContextKeyValue(requestHeaders, RequestResponseHeaders.RequestContextSourceKey, sourceApplicationId);
+                        }
+                    }
+
+                    // Add the root ID
+                    string rootId = telemetry.Context.Operation.Id;
+                    if (!string.IsNullOrEmpty(rootId) && !requestHeaders.Contains(RequestResponseHeaders.StandardRootIdHeader))
+                    {
+                        requestHeaders.Add(RequestResponseHeaders.StandardRootIdHeader, rootId);
+                    }
+
+                    // Add the parent ID
+                    string parentId = telemetry.Id;
+                    if (!string.IsNullOrEmpty(parentId) && !requestHeaders.Contains(RequestResponseHeaders.StandardParentIdHeader))
+                    {
+                        requestHeaders.Add(RequestResponseHeaders.StandardParentIdHeader, parentId);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Diagnostic event handler method for 'System.Net.Http.Response' event. Even in the case of an exception, this will still be called.
+        /// See https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L71 for more information.
+        /// </summary>
+        [DiagnosticName("System.Net.Http.Response")]
+        public void OnResponse(HttpResponseMessage response, Guid loggingRequestId)
+        {
+            if (response != null)
+            {
+                DependencyTelemetry telemetry;
```

Comment:
Have some uber exception handling so that anything that goes wrong in this OnResponse callback doesn't fail user's application. The user should still be able to live without dependency data going to AI.